### DISCUSSION TITLE
Does bun install support yarn workspaces?

### DISCUSSION THREAD

**Discussion by vijaybritto** [oven-sh/bun#456] [Q&A] [answered]
Does bun install support yarn workspaces. We have a monorepo that could use really fast installation times. I don't see any information on bun install docs about monorepo support. Does bun support this or plan to support it?

**Comment by fowled** [comment]
Was wondering that as well. It's the only thing that keeps me from switching.

**Comment by alirezabonab** [comment]
Same here, regarding the dependencies and packages management for a mono-repo project we are using `yarn workspaces`. it would be great to run `bun install` at the root of the project and it handles all nested projects in the mono-repo.

**Comment by Jarred-Sumner** [comment]
Not yet, but Bun will have support for workspaces

**Reply by mmmeff** [reply]
Awesome. Can't wait to make the switch!

**Reply by dcmwong** [reply]
Preventing me from switching too.  How do we track this feature?

**Reply by fowled** [reply]
For the moment, by following this discussion (I guess)

**Comment by janwilmake** [comment]
any update? and if I am using a mono repo , is there already a way to convert it to bun ?

**Comment by factoidforrest** [comment]
Excited for this one. A note on why this might be higher priority than you would think at first glance: 

Many more people who are plagued by slow package installs are using monorepos, much more than the average. This is the particular demographic that could really benefit from Bun's blindingly fast package management. Adding this feature could result in a lot of adoption.

**Comment by pleahmacaka** [comment]
Are there any alternatives I can use while waiting?

**Comment by Jarred-Sumner** [comment, marked as answer]
soon

edit: yes

https://github.com/oven-sh/bun/pull/1764

**Reply by claycoleman** [reply]
note, bun workspace support doesn't currently allow for wildcards in your `workspaces` setting

**Reply by claycoleman** [reply]
also note, bun doesn't currently support installing packages from an existing lockfile :/ this means if you're trying to switch to `bun install` for a monorepo, you'll have to accept one of these options:
- basically have your lockfile blown away and just deal with all package version changes
- manually (or write a script to) run `bun install` for each dependency with the version from your lockfile 😢 
- don't use bun for your monorepo until it supports this

see https://github.com/oven-sh/bun/issues/1751

**Reply by jatinsandilya** [reply]
Is this supported now after bun 1.0 ?

**Reply by andrejnsimoes** [reply]
Not yet @jatinsandilya. And multi-level globs are TBD too.

**Reply by Trosterud** [reply]
A year later in v1.1.27, is this supported? No specific docs on migrating from existing workspaces and what to do with existing lockfiles. But states:

`It's a standalone tool that will work in pre-existing Node.js projects; if your project has a package.json, bun install can help you speed up your workflow.`

and

`Glob support — Bun supports full glob syntax in "workspaces" (see [here](https://bun.sh/docs/api/glob#supported-glob-patterns) for a comprehensive list of supported syntax), except for exclusions (e.g. !**/excluded/**), which are not implemented yet.`

**Reply by Trosterud** [reply]
Also see thread on using synp
https://github.com/oven-sh/bun/issues/1751#issuecomment-2203399902
